client/src/components/reader/Section/ResourceViewer/Wrapper.js

```diff
@@ -0,0 +1,94 @@
+import React, { PureComponent, PropTypes } from 'react';
+import Viewer from './Viewer';
+import throttle from 'lodash/throttle';
+
+export default class ResourceViewerWrapper extends PureComponent {
+
+  static displayName = "ResourceViewer.Wrapper";
+
+  static propTypes = {
+    resources: PropTypes.array,
+    containerSize: PropTypes.number,
+    currentResources: PropTypes.array,
+    body: PropTypes.object
+  };
+
+  constructor() {
+    super();
+    this.state = {
+      availableResourceMarkers: [],
+      viewableResourceMarkers: []
+    };
+
+    this.updateAvailableResourceMarkers = this.updateAvailableResourceMarkers.bind(this);
+    this.updateViewableResourceMarkers = this.updateViewableResourceMarkers.bind(this);
+
+    this.throttledUpdateViewableResources = throttle(() => {
+      this.updateViewableResourceMarkers();
+    }, 200);
+  }
+
+  componentDidMount() {
+    // Run once on mount ?
+    this.updateAvailableResourceMarkers();
+    this.updateViewableResourceMarkers();
+
+    window.addEventListener('scroll', this.throttledUpdateViewableResources);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.throttledUpdateViewableResources);
+  }
+
+  componentDidUpdate(prevProps) {
+    // Update available resource markers if we have new props or state
+    if (prevProps !== this.props) {
```

Comment:
This check shouldn't be necessary. Because it's a pure component, I wouldn't expect it to update if props were the same.